src/main/java/edu/harvard/iq/dataverse/search/SolrQueryCreator.java

```diff
@@ -0,0 +1,177 @@
+package edu.harvard.iq.dataverse.search;
+
+import edu.harvard.iq.dataverse.FieldType;
+import edu.harvard.iq.dataverse.search.dto.CheckboxSearchField;
+import edu.harvard.iq.dataverse.search.dto.NumberSearchField;
+import edu.harvard.iq.dataverse.search.dto.SearchBlock;
+import edu.harvard.iq.dataverse.search.dto.SearchField;
+import edu.harvard.iq.dataverse.search.dto.TextSearchField;
+import org.apache.commons.lang.StringUtils;
+
+import javax.ejb.Stateless;
+import java.util.List;
+import java.util.stream.Collectors;
+
+
+/**
+ * Class used for creating solr query used in advanced search.
+ */
+@Stateless
+public class SolrQueryCreator {
```

Comment:
Probably we would have to support cases when user type multiple words in one field.
Now it would translate to `fieldName:word1 word2 AND fieldName2: ...`. Probably we don't want that as word2 will refer to default field (NOT fieldName) which is probably `_text_` field in our case.